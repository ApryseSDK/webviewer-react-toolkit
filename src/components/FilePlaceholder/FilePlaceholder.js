'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.FilePlaceholder = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importDefault(require('react'));
var FilePlaceholder = function(_a) {
  var className = _a.className,
    extension = _a.extension;
  var filePlaceholderClass = (0, classnames_1.default)(
    'ui__base ui__filePlaceholder',
    className,
  );
  var formattedExtension =
    extension && '.'.concat(extension.replace(/^\./, ''));
  return react_1.default.createElement(
    'div',
    { className: filePlaceholderClass },
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--thumbnail',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-sm',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-xs',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-df',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-lgx',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-lg',
    }),
    react_1.default.createElement('div', {
      className:
        'ui__filePlaceholder__block ui__filePlaceholder__block--line-df',
    }),
    formattedExtension
      ? react_1.default.createElement(
          'div',
          { className: 'ui__filePlaceholder__extension' },
          formattedExtension,
        )
      : undefined,
  );
};
exports.FilePlaceholder = FilePlaceholder;
//# sourceMappingURL=FilePlaceholder.js.map
