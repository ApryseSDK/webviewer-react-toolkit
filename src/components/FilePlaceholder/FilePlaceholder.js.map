{"version":3,"file":"FilePlaceholder.js","sourceRoot":"","sources":["FilePlaceholder.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,wDAAkC;AAa3B,IAAM,eAAe,GAA6B,UAAC,EAAwB;QAAtB,SAAS,eAAA,EAAE,SAAS,eAAA;IAC9E,IAAM,oBAAoB,GAAG,IAAA,oBAAU,EAAC,8BAA8B,EAAE,SAAS,CAAC,CAAC;IAEnF,IAAM,kBAAkB,GAAG,SAAS,IAAI,WAAI,SAAS,CAAC,OAAO,CAAC,KAAK,EAAE,EAAE,CAAC,CAAE,CAAC;IAE3E,OAAO,CACL,uCAAK,SAAS,EAAE,oBAAoB;QAClC,uCAAK,SAAS,EAAC,kEAAkE,GAAG;QACpF,uCAAK,SAAS,EAAC,gEAAgE,GAAG;QAClF,uCAAK,SAAS,EAAC,gEAAgE,GAAG;QAClF,uCAAK,SAAS,EAAC,gEAAgE,GAAG;QAClF,uCAAK,SAAS,EAAC,iEAAiE,GAAG;QACnF,uCAAK,SAAS,EAAC,gEAAgE,GAAG;QAClF,uCAAK,SAAS,EAAC,gEAAgE,GAAG;QAEjF,kBAAkB,CAAC,CAAC,CAAC,uCAAK,SAAS,EAAC,gCAAgC,IAAE,kBAAkB,CAAO,CAAC,CAAC,CAAC,SAAS,CACxG,CACP,CAAC;AACJ,CAAC,CAAC;AAlBW,QAAA,eAAe,mBAkB1B"}