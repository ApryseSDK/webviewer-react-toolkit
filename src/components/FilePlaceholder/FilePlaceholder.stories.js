'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var react_1 = tslib_1.__importDefault(require('react'));
var FilePlaceholder_1 = require('../FilePlaceholder');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
var addon_knobs_1 = require('@storybook/addon-knobs');
exports.default = {
  title: 'Components/FilePlaceholder',
  component: FilePlaceholder_1.FilePlaceholder,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(FilePlaceholder_1.FilePlaceholder, {
    extension: (0, addon_knobs_1.text)('extension', 'pdf'),
  });
};
exports.Basic = Basic;
//# sourceMappingURL=FilePlaceholder.stories.js.map
