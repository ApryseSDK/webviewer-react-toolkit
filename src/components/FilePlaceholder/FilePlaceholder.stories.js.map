{"version":3,"file":"FilePlaceholder.stories.js","sourceRoot":"","sources":["FilePlaceholder.stories.tsx"],"names":[],"mappings":";;;;AAAA,wDAA0B;AAC1B,sDAAqD;AACrD,kEAAiC;AACjC,sDAA8C;AAE9C,kBAAe,EAAE,KAAK,EAAE,4BAA4B,EAAE,SAAS,EAAE,iCAAe,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAEpG,IAAM,KAAK,GAAG,cAAM,OAAA,8BAAC,iCAAe,IAAC,SAAS,EAAE,IAAA,kBAAI,EAAC,WAAW,EAAE,KAAK,CAAC,GAAI,EAAxD,CAAwD,CAAC;AAAvE,QAAA,KAAK,SAAkE"}