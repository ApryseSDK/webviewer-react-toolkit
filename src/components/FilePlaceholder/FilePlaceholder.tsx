import classnames from 'classnames';
import React, { FC } from 'react';

export interface FilePlaceholderProps {
  /**
   * Classname of the placeholder wrapper.
   */
  className?: string;
  /**
   * The file extension to display on the placeholder.
   */
  extension?: string;
}

export const FilePlaceholder: FC<FilePlaceholderProps> = ({ className, extension }) => {
  const filePlaceholderClass = classnames('ui__base ui__filePlaceholder', className);

  const formattedExtension = extension && `.${extension.replace(/^\./, '')}`;

  return (
    <div className={filePlaceholderClass}>
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--thumbnail" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-sm" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-xs" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-df" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-lgx" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-lg" />
      <div className="ui__filePlaceholder__block ui__filePlaceholder__block--line-df" />

      {formattedExtension ? <div className="ui__filePlaceholder__extension">{formattedExtension}</div> : undefined}
    </div>
  );
};
