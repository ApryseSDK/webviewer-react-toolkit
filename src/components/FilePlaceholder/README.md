Placeholder for files. You can include an extension to add it to the bottom of
the placeholder. These should be used when you are unable to generate thumbnails
for a given file, for example an unsupported file type.
