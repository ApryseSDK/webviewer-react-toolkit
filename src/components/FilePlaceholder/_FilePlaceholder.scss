.ui__filePlaceholder {
  background-color: white;
  padding: 10px;
  height: 164px;
  width: 128px;
  position: relative;

  &__block {
    background-color: #ccc;

    &--thumbnail {
      width: 40px;
      height: 40px;
      float: left;
    }

    &--line-sm {
      margin-left: 50px;
      margin-top: 10px;
      height: 6px;
    }

    &--line-xs {
      margin-left: 50px;
      margin-top: 10px;
      width: 40px;
      height: 6px;
    }

    &--line-df {
      margin-top: 20px;
      width: 90px;
      height: 6px;
    }

    &--line-lg {
      margin-top: 10px;
      width: 90px;
      height: 6px;
    }

    &--line-lgx {
      margin-top: 10px;
      height: 6px;
    }
  }

  &__extension {
    position: absolute;
    color: #ccc;
    text-align: center;
    font-size: $font-size-large;
    right: 10px;
    left: 10px;
    bottom: 10px;
  }
}
