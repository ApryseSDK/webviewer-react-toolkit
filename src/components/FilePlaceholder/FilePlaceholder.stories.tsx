import React from 'react';
import { FilePlaceholder } from '../FilePlaceholder';
import readme from './README.md';
import { text } from '@storybook/addon-knobs';

export default { title: 'Components/FilePlaceholder', component: FilePlaceholder, parameters: { readme } };

export const Basic = () => <FilePlaceholder extension={text('extension', 'pdf')} />;
