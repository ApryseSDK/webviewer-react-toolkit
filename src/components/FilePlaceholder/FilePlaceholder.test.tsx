import { shallow } from 'enzyme';
import React from 'react';
import { FilePlaceholder } from '../FilePlaceholder';

describe('FilePlaceholder component', () => {
  it('renders its contents', () => {
    const filePlaceholder = shallow(<FilePlaceholder />);
    expect(filePlaceholder.find('.ui__filePlaceholder')).toHaveLength(1);
  });

  it('snapshot renders default filePlaceholder', () => {
    const filePlaceholder = shallow(<FilePlaceholder />);
    expect(filePlaceholder).toMatchSnapshot();
  });
});
