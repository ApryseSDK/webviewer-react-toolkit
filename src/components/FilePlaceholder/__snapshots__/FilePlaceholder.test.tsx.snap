// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePlaceholder component snapshot renders default filePlaceholder 1`] = `
<div
  className="ui__base ui__filePlaceholder"
>
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--thumbnail"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-sm"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-xs"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-df"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-lgx"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-lg"
  />
  <div
    className="ui__filePlaceholder__block ui__filePlaceholder__block--line-df"
  />
</div>
`;
